spring-boot-project/spring-boot-tools/spring-boot-configuration-processor/src/main/java/org/springframework/boot/configurationprocessor/ConfigurationMetadataAnnotationProcessor.java

```diff
@@ -353,6 +356,20 @@ private ConfigurationMetadata mergeAdditionalMetadata(ConfigurationMetadata meta
 		return metadata;
 	}
 
+	private void validateDefaultValuesResolved(ConfigurationMetadata metadata) {
+        // TODO do we want this? How do we configure annotation processor w/ 'levers'
```

Comment:
Ideally fail when we know that we're in an environment where default values can be discovered. Ignore otherwise.